### PR TITLE
Remove closures Convnext

### PR DESCRIPTION
**Pull request by IMvision12**
# What does this PR do?

fix : #1419
As convnext does not have weights yet, i have done comparison of total number of params [colab](https://colab.research.google.com/drive/11SSf4cHgOBBPinRchYdsJXTuXFm9Hour?usp=sharing)

@ianstenbit

### CODE REVIEW

**Comment by IMvision12** [comment]
Test file will be addressed by this PR #1389

File: keras_cv/models/convnext.py [hunk 1]
```diff
@@ -161,238 +157,266 @@ def ConvNeXtBlock(
...
-    def apply(inputs):
-        x = inputs
-
-        x = layers.Conv2D(
-            filters=projection_dim,
-            kernel_size=7,
-            padding="same",
-            groups=projection_dim,
-            name=name + "_depthwise_conv",
+    x1 = x
```
Comment (ianstenbit) [inline keras_cv/models/convnext.py:166]: nit: maybe let's call this x0 or "inputs"

File: keras_cv/models/convnext.py [hunk 2]
```diff
@@ -161,238 +157,266 @@ def ConvNeXtBlock(
...
+    if drop_path_rate:
+        layer = StochasticDepth(drop_path_rate, name=name + "_stochastic_depth")
+        return layer([x1, x])
+    else:
+        layer = layers.Activation("linear", name=name + "_identity")
+        return x1 + layer(x)
 
 
-def Head(num_classes, activation="softmax", name=None):
+def Head(x, num_classes, activation="softmax", name=None):
```
Comment (ianstenbit) [inline keras_cv/models/convnext.py:194]: Let's rename this to apply_head

File: keras_cv/models/convnext.py [hunk 3]
```diff
@@ -161,238 +157,266 @@ def ConvNeXtBlock(
...
+      classes: optional int, number of classes to classify images into (only
+            to be specified if `include_top` is `True`).
       classifier_activation: A `str` or callable. The activation function to use
-        on the "top" layer. Ignored unless `include_top=True`. Set
-        `classifier_activation=None` to return the logits of the "top" layer.
-      name: An optional name for the model.
-
+            on the "top" layer. Ignored unless `include_top=True`. Set
+            `classifier_activation=None` to return the logits of the "top" layer.
+      name: (Optional) name to pass to the model.  Defaults to "{name}".
```
Comment (ianstenbit) [inline keras_cv/models/convnext.py:251]: Defaults to "convnext"

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thank you! In your colab, can you also verify numerical equivalence for the two models?

(Just save the weights from one and load it into the other model and check that the outputs match)

**Comment by IMvision12** [comment]
> (Just save the weights from one and load it into the other model and check that the outputs match)
We don't have weights for convnext

**Comment by ianstenbit** [comment]
> > (Just save the weights from one and load it into the other model and check that the outputs match)
> > We don't have weights for convnext

Right -- randomly initialized weights are fine (just to verify that outputs match, they don't need to be good outputs)

**Comment by IMvision12** [comment]
Here : https://colab.research.google.com/drive/1UNLsPLS-lZrJ2mBv0PSJ_RiAHf8UVgt0?usp=sharing

**Comment by ianstenbit** [comment]
> Here : https://colab.research.google.com/drive/1UNLsPLS-lZrJ2mBv0PSJ_RiAHf8UVgt0?usp=sharing

Updated your Colab a bit to also perform numerical equivalence testing for a sentinel input:
https://colab.research.google.com/gist/ianstenbit/fff6fa172e5a50ebfc72ceab2adb5f02/copy-of-convnext.ipynb

Looks good.

**Comment by ianstenbit** [comment]
> Here : https://colab.research.google.com/drive/1UNLsPLS-lZrJ2mBv0PSJ_RiAHf8UVgt0?usp=sharing

Updated your Colab a bit to also perform numerical equivalence testing for a sentinel input:
https://colab.research.google.com/gist/ianstenbit/fff6fa172e5a50ebfc72ceab2adb5f02/copy-of-convnext.ipynb

Looks good.

**Review by ianstenbit — APPROVED** [review]
Looks good! 

I re-started the CI runs because they looked flaky. If the issue persists we can take another look.

**Pull request merged**